### PR TITLE
Optimization problem_10 in project_euler

### PR DESCRIPTION
**Pull request by shellhub**
### **Describe your change:**

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by shellhub** [comment]
@cclauss Please review. Thanks :)

**Review by AUNTB05 — APPROVED** [review]

**Review by AUNTB05 — APPROVED** [review]

**Review by AUNTB05 — APPROVED** [review]

**Review by AUNTB05 — APPROVED** [review]

**Comment by cclauss** [comment]
For optimizations it would be good to see a timeit benchmark that proves the performance gain.

**Comment by cclauss** [comment]
The most important tests to optimize are:
https://travis-ci.com/github/TheAlgorithms/Python/builds/185436069

**Review by AUNTB05 — APPROVED** [review]

**Review by AUNTB05 — APPROVED** [review]

**Comment by shellhub** [comment]
@cclauss `benchmark` function output is below:
``` bash
14.957698989999999
0.06171911999999935
```
It proved second version is more faster.

File: project_euler/problem_09/sol1.py [hunk 1]
```diff
@@ -30,6 +29,38 @@ def solution():
...
+def benchmark() -> None:
+    """
+    Benchmark code comparing two different version function.
+    """
+    import timeit
+
+    print(timeit.timeit("solution()", setup="from __main__ import solution", number=1))
+    print(
+        timeit.timeit(
+            "solution_fast()", setup="from __main__ import solution_fast", number=1
```
Comment (cclauss) [inline project_euler/problem_09/sol1.py:60]:
```suggestion
            "solution_fast()", setup="from __main__ import solution_fast", number=1000
```
Just to make sure it is not a fluke.

Comment (shellhub) [reply project_euler/problem_09/sol1.py]: If we replace number `1` with `1000`. It take a long time to complete.

File: project_euler/problem_09/sol1.py [hunk 2]
```diff
@@ -30,6 +29,38 @@ def solution():
...
+                return a * b * c
+
+
+def benchmark() -> None:
+    """
+    Benchmark code comparing two different version function.
+    """
+    import timeit
+
+    print(timeit.timeit("solution()", setup="from __main__ import solution", number=1))
```
Comment (shellhub) [inline project_euler/problem_09/sol1.py:57]:
```suggestion
    print(timeit.timeit("solution()", setup="from __main__ import solution", number=1000))
```

Comment (shellhub) [reply project_euler/problem_09/sol1.py:57]: We should also change this line ? @cclauss

Comment (shellhub) [inline project_euler/problem_09/sol1.py:57]:
```suggestion
    print(timeit.timeit("solution()", setup="from __main__ import solution", number=1000))
```

**Comment by shellhub** [comment]
@cclauss Done. Please review again.

**Review by cclauss — APPROVED** [review]

**Pull request merged**